### PR TITLE
Add Salesforce connector guide

### PR DESCRIPTION
**Pull request by pramodya1994**
## Purpose
Adding guide on using Salesforce connector.
Fixes; https://github.com/wso2/ballerina-integrator/issues/167

### CODE REVIEW

File: examples/saas-clients/salesforce/working-with-salesforce-client/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+target
+ballerina-internal.log
```
Comment (IrushiL) [inline examples/saas-clients/salesforce/working-with-salesforce-client/.gitignore]: New line?

File: examples/saas-clients/salesforce/working-with-salesforce-client/README.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+```
+working-with-salesforce-client
+    └── src
+         └── guide
+               └── tests
+               └── resources
+               └── ballerina.conf
+               └── working_with_salesforce_client.bal
+```
+- In the `working-with-salesforce-client/src/guide` directory there is a `ballerina.conf` file, which you should enter
```
Comment (IrushiL) [inline examples/saas-clients/salesforce/working-with-salesforce-client/README.md]:
```suggestion
- In the `working-with-salesforce-client/src/guide` directory there is a `ballerina.conf` file, where you should enter
```

File: examples/saas-clients/salesforce/working-with-salesforce-client/README.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+all the confugurations related to the guide. Add the following details you obtained above within the quotes.
```
Comment (IrushiL) [inline examples/saas-clients/salesforce/working-with-salesforce-client/README.md]:
```suggestion
all the configurations related to the guide. Add the following details you obtained above within the quotes.
```

File: examples/saas-clients/salesforce/working-with-salesforce-client/README.md [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+following operations:
+
+* Create Account with Contacts & Opportunities
+* Create a Contact for a given Account ID
+* Create a Opportunity for a given Account ID
+* Execute a SOQL query
+
+## Prerequisites
+- [Ballerina Distribution](https://ballerina.io/learn/getting-started/)
+- A Text Editor or an IDE
```
Comment (samgnaniah) [inline examples/saas-clients/salesforce/working-with-salesforce-client/README.md]: We should recommend only VS Code.

File: examples/saas-clients/salesforce/working-with-salesforce-client/README.md [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+```bash
+$ ballerina run working_with_amazons3_client.balx
+```
+
+### Testing
+
+- Navigate to `working-with-salesforce-client`, and execute the following command to start the service:
+
+```bash
+$ ballerina run working_with_amazons3_client.bal
```
Comment (samgnaniah) [inline examples/saas-clients/salesforce/working-with-salesforce-client/README.md]: Should we use `ballerina build` instead?

File: examples/saas-clients/salesforce/working-with-salesforce-client/README.md [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+
+<!-- INCLUDE_CODE: guide/src/working_with_salesforce_client.bal -->
+
+### Deployment
+
+#### Deploying locally
+You can deploy the services that you developed above in your local environment. You can create the Ballerina executable 
+archives (.balx) first as follows.
+
+**Building**
```
Comment (samgnaniah) [inline examples/saas-clients/salesforce/working-with-salesforce-client/README.md]: I feel this should be merged with the Testing section. Deployment should be primarily focused on how to deploy.

File: examples/saas-clients/salesforce/working-with-salesforce-client/README.md [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+
+#### Developing the service
+
+The following code is the completed sample which exposes the following services:
+- createAccount: Creates a new Account and related Contacts & Opportunities using the given JSON.
+- createContact: Creates a new Contact using given Contact details.
+- createOpportunity: Creates a new Opportunity using given Opportunity details.
+- executeQuery: Execute a given SOQL query and return results as a JSON.
+
+<!-- INCLUDE_CODE: guide/src/working_with_salesforce_client.bal -->
```
Comment (samgnaniah) [inline examples/saas-clients/salesforce/working-with-salesforce-client/README.md]: Shall we consider breaking this into smaller sections for clarity?

File: examples/saas-clients/salesforce/working-with-salesforce-client/README.md [hunk 8]
```diff
@@ -0,0 +1,256 @@
+# Working with Salesforce client
+
+## Overview
+The Ballerina Salesforce client enables you to perform CRUD operations for SObjects, query using SOQL, search using 
+SOSL and describe SObjects and organizational data through the Salesforce REST API.
+
+This example explains how to use the Salesforce client to connect with the Salesforce instance and perform the 
+following operations:
+
+* Create Account with Contacts & Opportunities
```
Comment (samgnaniah) [inline examples/saas-clients/salesforce/working-with-salesforce-client/README.md]: We need a "What you'll build" type section with a diagram here.

**Comment by pramodya1994** [comment]
Add another PR https://github.com/wso2/ballerina-integrator/pull/384 to implement this guide to resolve directory conflicts.

**Pull request closed without merging**